src/java/org/apache/cassandra/cql3/statements/TruncateStatement.java

```diff
@@ -66,7 +68,10 @@ public ResultMessage execute(QueryState state, QueryOptions options, long queryS
                 throw new InvalidRequestException("Cannot TRUNCATE materialized view directly; must truncate base table instead");
 
             if (metaData.isVirtual())
-                throw new InvalidRequestException("Cannot truncate virtual tables");
+            {
+                executeForVirtualTable(metaData.id);
+                return null;
```

Comment:
nit: In this case I would have use a `if-else` to keep a single return. 

I did not remember what we normally return for TRUNCATE statements and I had to check that it was matching the other one.

Reply:
Done.